### PR TITLE
Use [[maybe_unused]] instead of __attribute__((unused))

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
closes https://github.com/maplibre/maplibre-native/issues/906

### CODE REVIEW

**Comment by SiarheiFedartsou** [comment]
@louwers there is a compilation error in this PR, I will fix it later :)

File: test/storage/offline_database.test.cpp [hunk 1]
```diff
@@ -50,7 +45,7 @@ static FixtureLog::Message error(ResultCode code, const char* message) {
     return {EventSeverity::Error, Event::Database, static_cast<int64_t>(code), message};
 }
 
-static MBGL_UNUSED FixtureLog::Message warning(ResultCode code, const char* message) {
```
Comment (SiarheiFedartsou) [inline test/storage/offline_database.test.cpp:53]: Well, it seems `[[maybe_unused]]` cannot be used to suppress unused warnings for functions 🤷🏻

Comment (louwers) [reply test/storage/offline_database.test.cpp:53]: Which compiler complains?

Comment (louwers) [reply test/storage/offline_database.test.cpp:53]: Ah

```
/Users/ec2-user/actions-runner/_work/maplibre-native/maplibre-native/test/storage/offline_database.test.cpp:47:10: error: 'maybe_unused' attribute cannot be applied to types
static [[maybe_unused]] FixtureLog::Message warning(ResultCode code, const char* message) {
```

**Pull request closed without merging**